test/generate_prim_sig_tests.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python
+
+from argparse import ArgumentDefaultsHelpFormatter, ArgumentParser
+import numbers
+import sys
+
+from sig_utils import handle_function_list, get_signatures
+from signature_parser import SignatureParser
+from code_generator import CodeGenerator
+
+src_folder = "./test/expressions/"
+build_folder = "./test/expressions/"
+
+test_code_template = """
+TEST(PrimSigsTests, {test_name}) {{
+/*
+{comment}
+ */    
+{code}
+{check}
+}}
+"""
+
+
+def main():
+    """
+    Generates tests to check primitive scalar functions.
+
+    The tests check whether the primitive scalar functions are defined in the
+    stan::math namespace and that no stan::math::var objects are created when
+    they are used. If a var object is created, some of the input arguments
+    were cast into a var and the reverse mode function was called.
+
+    """
+
+    tests = []
+
+    default_checks = True
+    signatures = set(get_signatures())
+
+    for n, signature in enumerate(signatures):
+        sp = SignatureParser(signature)
+
+        # skip ignored signatures if running the default checks
+        if default_checks and sp.is_ignored():
+            continue
+
+        # skip signatures without inputs that can be eigen types
+        if (
+            not sp.has_scalar_only_args()
+            or sp.returns_complex()
+            or sp.is_nullary()
+            or sp.is_rng()
+        ):
+            continue
+
+        cg = CodeGenerator()
+
+        arg_list_no_expression = cg.build_arguments(
+            sp, sp.number_arguments() * ["Prim"], size=1
+        )
+
+        # Check results with expressions and without are the same
+        cpp_function_name = "stan::math::" + sp.function_name
+        result_1 = cg.function_call_assign(cpp_function_name, *arg_list_no_expression)
+        result_2 = cg.function_call_assign(cpp_function_name, *arg_list_no_expression)
```

Comment:
`result_1` and `result_2` have the same definition.  Should one of these have a different arg list?